pkg/cmd/roachtest/tests/backup.go

```diff
@@ -164,58 +171,338 @@ func registerBackupNodeShutdown(r registry.Registry) {
+		// https://github.com/cockroachdb/cockroach/pull/79666 is backported to
+		// 21.2 and the mixed version map for roachtests is bumped to the 21.2
+		// patch release with the backport.
+		//
+		// The bug above means that nodes for which we have disabled adoption may
+		// still lay claim on the job, and then not clear their claim on realizing
+		// that adoption is disabled. To prevent the job from getting wedged, we
+		// manually clear the claim session on the job for instances where job
+		// adoption is disabled. This will allow other node's in the cluster to
+		// adopt the job and run it.
+		removeJobClaimsForNodes(ctx, t, db, nodesWithAdoptionDisabled, jobID)
+
+		var status string
+		var payloadBytes []byte
+		err := db.QueryRow(`SELECT status, payload FROM system.jobs WHERE id = $1`, jobID).Scan(&status, &payloadBytes)
+		require.NoError(t, err)
+		if jobs.Status(status) == jobs.StatusFailed {
+			payload := &jobspb.Payload{}
+			if err := protoutil.Unmarshal(payloadBytes, payload); err == nil {
+				t.Fatalf("job failed: %s", payload.Error)
+			}
+			t.Fatalf("job failed")
+		}
+		if e, a := expectedStatus, jobs.Status(status); e != a {
+			return errors.Errorf("expected job status %s, but got %s", e, a)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// fingerprint returns a fingerprint of `db.table`.
+func fingerprint(ctx context.Context, conn *gosql.DB, db, table string) (string, error) {
+	var b strings.Builder
+
+	query := fmt.Sprintf("SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE %s.%s", db, table)
+	rows, err := conn.QueryContext(ctx, query)
+	if err != nil {
+		return "", err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var name, fp string
+		if err := rows.Scan(&name, &fp); err != nil {
+			return "", err
+		}
+		fmt.Fprintf(&b, "%s: %s\n", name, fp)
+	}
+
+	return b.String(), rows.Err()
+}
+
+// verifyBackupStep compares the backed up and restored table fingerprints and
+// ensures they're the same.
+func verifyBackupStep(
+	node option.NodeListOption, backupLoc string, dbName, tableName, intoDB string,
+) versionStep {
+	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
+		db := u.conn(ctx, t, node[0])
+
+		// First fingerprint the original table.
+		origFingerprint, err := fingerprint(ctx, db, dbName, tableName)
+		require.NoError(t, err)
+
+		// Now, restore the backup.
+		_, err = db.ExecContext(ctx, fmt.Sprintf(`CREATE DATABASE %s`, intoDB))
+		require.NoError(t, err)
+		_, err = db.ExecContext(ctx, fmt.Sprintf(`RESTORE TABLE %s.%s FROM LATEST IN '%s' WITH into_db = '%s'`,
+			dbName, tableName, backupLoc, intoDB))
+		require.NoError(t, err)
+
+		restoredFingerPrint, err := fingerprint(ctx, db, intoDB, tableName)
+		require.NoError(t, err)
+		if origFingerprint != restoredFingerPrint {
+			t.Fatal("expected backup and restore fingerprints to match")
+		}
+	}
+}
+
 func registerBackupMixedVersion(r registry.Registry) {
+	setShortJobIntervalsStep := func(node int) versionStep {
```

Comment:
I think it's worth adding a comment explaining why this function is necessary

Reply:
done.